atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -97,122 +102,65 @@ def run(self):
+        image_name = image.to_str(tag=False)
+        new_image = '{}@sha256:{}'.format(image_name, digest)
+        return ImageName.parse(new_image)
 
+    def _store_manifest_digest(self, image):
+        """Store manifest list or v2 schema 2 manifest digest """
+        image_str = image.to_str()
         manifest_list = self._get_manifest_list(image)
-
-        if not manifest_list:
-            self.log.warning(
-                "Empty manifest list for image %s. Collected image digests will be "
-                "incomplete and may cause race conditions during build", image
-            )
-            return
-
-        manifest_list_dict = manifest_list.json()
-
-        try:
-            arch_to_platform = get_goarch_to_platform_mapping(self.workflow)
-        except KeyError:
-            self.log.warning('Cannot collect platforms digests for parent images '
-                             'because platform descriptors are not defined')
-            return
-
-        for manifest in manifest_list_dict['manifests']:
-            arch = manifest['platform']['architecture']
+        if manifest_list:
+            manifest_digest = get_checksums(BytesIO(manifest_list.content), ['sha256'])
+            media_type = get_manifest_media_type('v2_list')
+        else:
+            digests_dict, _ = get_all_manifests(image, image.registry,
+                                                self.parent_registry_insecure,
+                                                self.parent_registry_dockercfg_path,
+                                                versions=('v2'))
+            media_type = get_manifest_media_type('v2')
             try:
-                present_platform = arch_to_platform[arch]
+                manifest_digest = digests_dict['v2']
             except KeyError:
-                self.log.warning("Cannot map architecture='{}' to platform".format(arch))
-                continue
-            else:
-                digest = manifest['digest']
-                self.log.info("Collecting digest for image '%s' ('%s'): '%s'",
-                              image, present_platform, digest)
-                parents_digests.update_image_digest(
-                    image, present_platform, digest
-                )
-
-    def _get_image_for_different_arch(self, image, platform):
-        """Get image from random arch
-
-        This is a workaround for aarch64 platform, because orchestrator cannot
-        get this arch from manifests lists so we have to provide digest of a
-        random platform to get image metadata for orchestrator.
-
-        For standard platforms like x86_64, ppc64le, ... this method returns
-        the corresponding digest
-
-        """
-        parents_digests = self.workflow.builder.parent_images_digests
-        try:
-            digests = parents_digests.get_image_digests(image)
-        except KeyError:
-            return None
+                raise RuntimeError('Could not extract manifest list or '
```

Comment:
instead of raising exception in this method, it should collect all errors for all missing parents, and raise exception in run() at the end

Reply:
done